### PR TITLE
Avoid crash caused by KDE qqc2-desktop-style

### PR DESCRIPTION
**Pull request by StefanBruens**
Workaround/fix for #5488
Also see https://bugs.kde.org/show_bug.cgi?id=417900

### CODE REVIEW

File: cura_app.py [hunk 1]
```diff
@@ -207,5 +207,11 @@ def exceptHook(hook_type, value, traceback):
     import trimesh.exchange.load
     os.environ["LD_LIBRARY_PATH"] = old_env
 
+# WORKAROUND: CURA-5488
```
Comment (nallath) [inline cura_app.py]:
```suggestion
# WORKAROUND: Cura#5488 
```
CURA-number is used for our Jira tickets (which is an internal tracker). Github uses [reponame]#[issue_nr]

**Comment by thopiekar** [comment]
Additionally, the fix shall be KDE-specific but is applied on every running Linux system.
Don't like my environment workaround/hack in my packaging files, too, but everything that goes into Cura here must be implemented perfectly in my opinion.

**Comment by Ghostkeeper** [comment]
The default style shouldn't be used anyway, as everything should either be Cura-themed or system-themed. But of course there might still be fields that we don't specialise because they normally don't influence things - like inner margins of the slider handles or something - that make KDE crash. If they don't affect the layout of Cura I'm fine with accepting this change for all Linux distributions, or even all platforms. If it *does* affect the layout of Cura, we have a different problem on our hands that apparently the layout is not fully themed.

**Comment by Ghostkeeper** [comment]
It does break a lot of stuff though :cry:

image

**Comment by credomane** [comment]
This patch doesn't quite fix everything. Importing a STL file then importing a second one will actually import the first stl again and the second stl file. Also drop down's are misaligned but not a bit deal. At least the crashing is resolved with this patch.

I use the following snippet to fully fix Cura on KDE. Only bad part is the open file dialog isn't the fancy KDE one anymore and instead a minimal one. Everything works as expected and looks like it should. I actually forgot I applied my snipped below until I updated to Cura 4.5 just an hour ago and had to find it again.

```python
if Platform.isLinux():
    os.environ.pop("DESKTOP_SESSION", None)
    os.environ.pop("KDE_FULL_SESSION", None)
    os.environ.pop("XDG_CURRENT_DESKTOP", None)
```

**Comment by protist** [comment]
FWIW a similar patch is mentioned [here](https://bugs.archlinux.org/task/62133).

**Comment by Ghostkeeper** [comment]
I don't think we can merge this as it is now, since it seems to break the layout for other systems.

**Comment by onitake** [comment]
@Ghostkeeper Can you be more specific how this breaks other systems?
Right now, Cura is practically unusable on many Linux systems because of this rendering issue. A fix is urgently needed.

**Comment by Ghostkeeper** [comment]
I posted a screenshot of it there:

Screenshot

The things I can see on the main screen are:
* There are now margins between the setting category headers.
* Margins above and below every action button is not big enough.
* Drop-downs become too slim.
* Open file button too small.
* There's a red line on the active extruder tab.
* There are margins on the tool bar on the left.

File: cura_app.py (hunk 1 above)
Comment (StefanBruens) [inline cura_app.py:210]:
```suggestion
# WORKAROUND: Cura#5488
```

Comment (StefanBruens) [reply cura_app.py]: Applied without extra whitespace

File: cura_app.py [hunk 2]
```diff
@@ -207,5 +207,11 @@ def exceptHook(hook_type, value, traceback):
     import trimesh.exchange.load
     os.environ["LD_LIBRARY_PATH"] = old_env
 
+# WORKAROUND: Cura#5488
+# When using the KDE qqc2-desktop-style, the UI layout is completely broken, and
+# even worse, it crashes when switching to the "Preview" pane.
+if Platform.isLinux():
+    os.environ["QT_QUICK_CONTROLS_STYLE"] = "material"
```
Comment (StefanBruens) [inline cura_app.py:214]:
```suggestion
    os.environ["QT_QUICK_CONTROLS_STYLE"] = "default"
```

**Comment by StefanBruens** [comment]
@Ghostkeeper -- please check again:

Screenshot_20201208_145708

**Comment by Ghostkeeper** [comment]
Yeah, that was it. It looks normal now to me.

**Pull request merged**